rpc/src/server/api.rs

```diff
@@ -117,6 +118,20 @@ impl api_server::Api for RpcApi {
     ) -> Result<Response<SubmitProvenTransactionResponse>, Status> {
         debug!(target: COMPONENT, request = ?request.get_ref());
 
+        let request = request.into_inner();
+
+        let tx = ProvenTransaction::read_from_bytes(&request.transaction)
+            .map_err(|_| Status::invalid_argument("Invalid transaction"))?;
+
+        let tx_verifier = TransactionVerifier::new(96);
+
+        tx_verifier.verify(tx.clone()).map_err(|_| {
+            Status::invalid_argument(format!(
+                "Invalid transaction proof for transaction: {}",
+                tx.id().to_hex()
+            ))
```

Comment:
with the latest changes in `miden-base`, we can probably get rid of `to_hex()` part.